src/proof/dot/dot_printer.h

```diff
@@ -99,6 +194,12 @@ class DotPrinter
 
   /** Counter that indicates the current rule ID */
   uint64_t d_ruleID;
+
+  /** Vector that holds all the scopes args */
+  std::vector<std::reference_wrapper<const std::vector<Node>>> d_scopesArgs;
+
+  /** Array with all the subgraphs description strings */
+  std::ostringstream* d_subgraphsStr;
```

Comment:
I think it's simpler to just do an std::vector with this guy and you add elements to it in the loop in the constructor.